include/revng-c/mlir/Dialect/Clift/IR/CliftAttributes.h

```diff
@@ -42,40 +42,40 @@ class StructType
 public:
   using Base::Base;
 
-  static StructType get(MLIRContext *ctx, uint64_t ID);
+  static StructType
+  get(MLIRContext *Context, uint64_t ID, llvm::StringRef Name, uint64_t Size);
 
   static StructType
   getChecked(llvm::function_ref<InFlightDiagnostic()> EmitError,
-             MLIRContext *ctx,
-             uint64_t ID);
+             MLIRContext *Context,
+             uint64_t ID,
+             llvm::StringRef Name,
+             uint64_t Size);
```

Comment:
Also, why do we need the `Size`? I think that the `Size` should always be known, not something that is attached afterwards when the struct is completed.

Reply:
That is correct. This is the function which does not take the field array and produces a potentially incomplete attribute.